### ISSUE TITLE
max-lines-per-function ignores arrow IIFEs

### ISSUE THREAD

**Issue by cherryblossom000** [eslint/eslint#13332]
**Tell us about your environment**
* **ESLint Version:** 7.0.0
* **Node Version:** 12.16.2
* **npm Version:** 6.14.4

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```json
{
  "env": {
    "es6": true
  },
  "rules": {
    "max-lines-per-function": [2, 1]
  }
}

```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
(() => {
})()
```

```bash
eslint test.js
```

[Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG1heC1saW5lcy1wZXItZnVuY3Rpb246IFsyLCAxXSAqL1xuKCgpID0+IHtcbn0pKCkiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

**What did you expect to happen?**
No errors because `max-lines-per-function` does not include any code in IIFEs by default.

**What actually happened? Please include the actual, raw output from ESLint.**
```
/path/to/test.js
  1:2  error  Arrow function has too many lines (2). Maximum allowed is 1  max-lines-per-function

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**
Yes.

[lib/rules/max-lines-per-function.js:137](https://github.com/eslint/eslint/blob/master/lib/rules/max-lines-per-function.js#L137)
```diff
-            return node.type === "FunctionExpression" && node.parent && node.parent.type === "CallExpression" && node.parent.callee === node;
+            return (node.type === "FunctionExpression" || node.type === "ArrowFunctionExpression") && node.parent && node.parent.type === "CallExpression" && node.parent.callee === node;
```

**Comment by yeonjuan** [comment]
@cherryblossom000 
Thanks for the reporting! It seems a bug to me.
